pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -425,7 +425,39 @@ func TestRhpamProdImmutableEnvironment(t *testing.T) {
 	assert.Nil(t, err, "Error getting prod environment")
 
 	assert.True(t, env.SmartRouter.Omit, "SmarterRouter should be omitted")
+	assert.True(t, env.Console.Omit, "Console should be omitted by default on immutable env.")
 	assert.Equal(t, "test-rhpamcentrmon", env.Console.DeploymentConfigs[0].ObjectMeta.Name)
+	assert.Equal(t, int32(0), env.Console.DeploymentConfigs[0].Spec.Replicas)
+	assert.Equal(t, bcmImage+":"+cr.Status.Applied.Version, env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Image)
+}
+
+func TestRhpamProdImmutableEnvironmentWithConsole(t *testing.T) {
+	replicas := int32(2)
+
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProductionImmutable,
+			Objects: api.KieAppObjects{
+				Console: api.ConsoleObject{
+					KieAppObject: api.KieAppObject{
+						Replicas: &replicas,
```

Comment:
can we add another test where you set replicas to 0 and test for Omit = true? Also, can you please test run this in a live cluster to ensure nothing with the other envs break?

Reply:
hmm... wait, ignore that new unit test. it won't work and doesn't fit here. but def make sure envs work as expected.

Reply:
I did run only the immutable on ocp, let me try the others one.